### DISCUSSION TITLE
definePageMeta in module's page

### DISCUSSION THREAD

**Discussion by Stiropor** [nuxt/nuxt#15881] [Questions] [answered]
I'm making a module which also has pages. Pages are registered and working fine:

```javascript
// Auto register pages
extendPages((pages) => {
	pages.push({
		name: 'bialconfigurator-configurator',
		path: '/configurator',
		file: resolve(__dirname, './pages/configurator.vue')
	})
})
```
But if I try to change the layout inside this page (having a configurator.vue inside nuxt's layouts directory):
```javascript
<template>
	<div>
		<Configurator />
	</div>
</template>
<script setup lang="ts">
definePageMeta({
	layout: 'configurator'
})
</script>
```

I get a `definePageMeta() is a compiler-hint helper that is only usable inside the script block of a single file component which is also a page. Its arguments should be compiled away and passing it at runtime has no effect.
` in my console. It works fine, but I'm getting this notification.

Why? This is a page, it is a script. How can I change the layout in module's page?

**Comment by ndragun92** [comment, marked as answer]
I think u need one more thing since u are overriding its meta. So try this

```javascript
extendPages((pages) => {
	pages.push({
		name: 'bialconfigurator-configurator',
		path: '/configurator',
		meta: {layout: 'configurator'},
		file: resolve(__dirname, './pages/configurator.vue')
	})
})
```

Hope it helps. Maybe syntax wise its not 100% accurate but I think it is associated with meta in pages

**Comment by Stiropor** [comment]
@ndragun92 Thanks, this works as intended 🚀  But it's weird I need to declare that in module instead of page.

**Reply by ndragun92** [reply]
I think it happens only because if u don't define even an empty meta as an object it will break the whole implementation
